### PR TITLE
Tests/enrich env model

### PR DESCRIPTION
**Pull request by MikeSquall**
Related to #10

### CODE REVIEW

File: tests/register.feature [hunk 1]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+         |{}             |
+         |{ test: 'test'}|
+         |{ services: [] }|
+  Then    Validation error 'env serviceName should be a string' is returned
+
+Scenario: Calling register method with a valid env, env method value comply with the model.  
+  Given   Environment Registry with register method
+  When    User calls register method by providing a valid envKey
+  And     Env model is in the correct structure
+  But     <myPropName> object property is not valid
```
Comment (MikeSquall) [inline tests/register.feature]:
```suggestion
  But     methods object <property> is not valid
```

File: tests/register.feature [hunk 2]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+         |myPropName     |
```
Comment (MikeSquall) [inline tests/register.feature]:
```suggestion
         |<property>     |
```

File: tests/register.feature [hunk 3]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+Scenario: Calling register method with a valid env, invalid env serviceName.
+  Given   Environment Registry with register method
+  When    User calls register method by providing a valid envKey
+  And     Env model is in the correct structure
+  But     User provides the following values for <serviceName>
+         |serviceName    |
+         |null           |
+         |undefined      |
+         |123            |
+         |'test'         |
```
Comment (MikeSquall) [inline tests/register.feature]: Remove

File: tests/register.feature [hunk 4]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+         |[]             |
+         |['test']       |
+         |{}             |
+         |{ test: 'test'}|
+         |{ services: [] }|
```
Comment (MikeSquall) [inline tests/register.feature]: Remove

File: tests/register.feature [hunk 5]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+Scenario: Calling register method with a valid env, invalid env serviceUrl.
+  Given   Environment Registry with register method
+  When    User calls register method by providing a valid envKey
+  And     Env model is in the correct structure
+  But     User provides the following values for <serviceUrl>
+         |serviceUrl    |
+         |null           |
+         |undefined      |
+         |123            |
+         |'test'         |
```
Comment (MikeSquall) [inline tests/register.feature]: Remove

File: tests/register.feature [hunk 6]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+         |[]             |
+         |['test']       |
+         |{}             |
+         |{ test: 'test'}|
+         |{ services: [] }|
```
Comment (MikeSquall) [inline tests/register.feature]: Remove

File: tests/register.feature [hunk 7]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+  And     Env model is in the correct structure
+  But     User provides the following values for <serviceMethods>
+         |serviceMethods |
+         |null           |
+         |undefined      |
+         |123            |
+         |'test'         |
+         |[]             |
+         |['test']       |
+         |{}             |
```
Comment (MikeSquall) [inline tests/register.feature]: Remove (this case is valid)

File: tests/register.feature [hunk 8]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+         |{ test: 'test'}|
+         |{ services: [] }|
```
Comment (MikeSquall) [inline tests/register.feature]: Remove

File: tests/register.feature [hunk 9]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+         |myPropName     |
+         |null           |
+         |undefined      |
+         |123            |
+         |'test'         |
+         |[]             |
+         |['test']       |
+         |{}             |
+         |{ test: 'test'}|
+         |{ services: [] }|
```
Comment (MikeSquall) [inline tests/register.feature]: Remove

File: tests/register.feature [hunk 10]
```diff
@@ -30,9 +30,77 @@ Scenario: Calling register method with invalid env
...
+  Then    Validation error 'env serviceName should be a string' is returned
```
Comment (stephanebenayoun) [inline tests/register.feature]: Need to be updated (and other occurrences also)

**Review by IrinaComindaru — APPROVED** [review]

**Pull request merged**